src/bevent.c

```diff
@@ -674,31 +704,34 @@ void bevent_unregister(bevent_h *eh)
  * @param call Call object (optional)
  * @param fmt  Formatted arguments
  * @param ...  Variable arguments
+ *
+ * @deprecated Use one of the event_xxx_emit() functions
  */
 void ua_event(struct ua *ua, enum ua_event ev, struct call *call,
 	      const char *fmt, ...)
 {
-	struct le *le;
 	char buf[256];
 	va_list ap;
+	static int w = CNT_DEPRECATED_WARNINGS;
 
 	va_start(ap, fmt);
 	(void)re_vsnprintf(buf, sizeof(buf), fmt, ap);
 	va_end(ap);
 
-	/* send event to all clients */
-	le = ehl.head;
-	while (le) {
-		struct ua_eh *eh = le->data;
-		le = le->next;
+	ua_event_private(ua, ev, call, buf);
 
-		if (call_is_evstop(call)) {
-			call_set_evstop(call, false);
-			break;
-		}
+	if (w && w--)
+		warning("Used deprecated ua_event() for %s. "
+			"Use one of event_xxx_emit() instead!\n",
+			uag_event_str(ev));
+	struct bevent event = {ev, buf, 0, false, { 0 } };
```

Comment:
Better construct with designated initialisers, this ensures non provided fields are zeroed by default and the struct can easily extended without touching all constructors.